backend/Squirrel.ConsoleApp/Extensions/HubConnectionExtensions.cs

```diff
@@ -0,0 +1,220 @@
+{
+    private static bool CurrentClientIsTarget(string currentClientId, string clientId)
+    {
+        if (string.IsNullOrEmpty(currentClientId))
+        {
+            return false;
+        }
+
+        if (!clientId.Equals(currentClientId))
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    public static void RegisterActions(this HubConnection hubConnection, string currentClientId, IApplicationBuilder app)
+    {
+        
+        hubConnection.On("TestExecuteQueryAsync", (string clientId, string filterName, int filterRowsCount) =>
+        {
+            if (!CurrentClientIsTarget(currentClientId, clientId))
+            {
+                return;
+            }
+
+            var getActionsService = app.ApplicationServices.GetRequiredService<IGetActionsService>();
+
+            hubConnection.InvokeAsync("TestReceiveExecutedQueryAsync", clientId, getActionsService.TestExecuteQueryAsync(filterName, filterRowsCount).Result);
+        });
+
+        hubConnection.On("GetAllTablesNamesAsync", (string clientId) =>
+        {
+            if (!CurrentClientIsTarget(currentClientId, clientId))
+            {
+                return;
+            }
+            
+            var getActionsService = app.ApplicationServices.GetRequiredService<IGetActionsService>();
+
+            hubConnection.InvokeAsync("ReceiveAllTablesNamesAsync", clientId, getActionsService.GetAllTablesNamesAsync().Result);
+        });
+
+        hubConnection.On("GetTableDataAsync", (string clientId, string tableName, int rowsCount) =>
+        {
+            if (!CurrentClientIsTarget(currentClientId, clientId))
+            {
+                return;
+            }
+
+            var getActionsService = app.ApplicationServices.GetRequiredService<IGetActionsService>();
+
+            hubConnection.InvokeAsync("ReceiveTableDataAsync", clientId, getActionsService.GetTableDataAsync(tableName, rowsCount).Result);
+        });
+
+        hubConnection.On("GetAllStoredProceduresNamesAsync", (string clientId) =>
+        {
+            if (!CurrentClientIsTarget(currentClientId, clientId))
+            {
+                return;
+            }
+
+            var getActionsService = app.ApplicationServices.GetRequiredService<IGetActionsService>();
+
+            hubConnection.InvokeAsync("ReceiveAllStoredProceduresNamesAsync", clientId, getActionsService.GetAllStoredProceduresNamesAsync().Result);
+        });
+
+        hubConnection.On("GetStoredProcedureDefinitionAsync", (string clientId, string storedProcedureName) =>
+        {
+            if (!CurrentClientIsTarget(currentClientId, clientId))
+            {
+                return;
+            }
+
+            var getActionsService = app.ApplicationServices.GetRequiredService<IGetActionsService>();
+
+            hubConnection.InvokeAsync("ReceiveStoredProcedureDefinitionAsync", clientId, getActionsService.GetStoredProcedureDefinitionAsync(storedProcedureName).Result);
```

Comment:
I guess all this functions can be done in more generic way, too many code repeat

Reply:
Maybe think about dictionary

Reply:
In a case when we use Client Id to identify the target of notification we don't need additional checks anymore and I have already deleted them.